urbit/lib/volt-json.hoon

```diff
@@ -0,0 +1,161 @@
+    ==
+  ::
+  ++  psbt
+    |=  jon=json
+    ^-  psbt:^psbt
+    ?.  ?=([%s @t] jon)
+      ~|  "%volt: invalid psbt"
+        !!
+    (need (from-base64:create:^psbt +.jon))
+  ::
+  ++  provider
+    |%
+    ++  command
+      |=  jon=json
+      ^-  command:provider:volt
+    %.  jon
+    %-  of
+    :~  set-url+so
+    ==
+  --
+  ++  ship  (su ;~(pfix sig fed:ag))
+  ++  network  (su (perk %main %testnet %regtest ~))
+  --
+::
+++  enjs
+  =,  enjs:format
+  |%
+  ++  provider
+    |%
+    ++  status
+      |=  s=status:provider:volt
+      ^-  json
+      (frond 'connected' b+=(s %connected))
+    ::
+    ++  update
+      |=  upd=update:provider:volt
+      ^-  json
+      ?-  -.upd
+        %res
+          ?+  +<.upd  !!
+            %node-info
+          (frond 'node-info' ~)
+          ==
+        %err
+          ?-  +<.upd
+            %rpc-error
+          (frond 'error' s+'rpc-error')
+            %not-connected
+          (frond 'error' s+'not-connected')
+            %bad-request
+          (frond 'error' s+'not-connected')
+          ==
+      ==
+  --
+  ++  update
+    |=  upd=update:volt
+    ^-  json
+    ?+    -.upd  (frond 'type' s+'unimplemented')
+        %channel-state
+      %-  pairs
+      :~  ['type' s+'channel-state']
+          ['id' s+`@t`(scot %ud chan-id.upd)]
+          ['status' s+chan-state.upd]
+      ==
+    ::
+        %new-invoice
+      %-  pairs
+      :~  ['type' s+'new-invoice']
+          ['payment-request' (payment-request payment-request.upd)]
+      ==
+    ::
+        %new-channel
+      %-  pairs
+      :~  ['type' s+'new-channel']
+          ['chan-info' (chan-info chan-info.upd)]
+      ==
+    ::
+        %channel-deleted
+      %-  pairs
+      :~  ['type' s+'channel-deleted']
+          ['id' s+`@t`(scot %ud id.upd)]
+      ==
+    ::
+        %initial-state
+      %-  pairs
+      :~  ['type' s+'initial-state']
+          ['chans' a+(turn chans.upd chan-info)]
+          ['txs' a+(turn txs.upd pay-info)]
+          ['invoices' a+(turn invoices.upd payment-request)]
+      ==
+    ==
+  ::
+  ++  chan-info
+    |=  info=chan-info:volt
+    ^-  json
+    %-  pairs
+    :~  ['id' s+`@t`(scot %ud id.info)]
+        ['who' (ship who.info)]
+        ['our' (numb our.info)]
+        ['his' (numb his.info)]
+        ['funding-address' (funding-address funding-address.info)]
+        ['status' s+status.info]
+    ==
+  ::
+  ++  funding-address
+    |=  f=(unit address:bitcoin)
+    ^-  json
+    ?~  f  ~
+    ?-   +<.f
+      %base58
+    !!
+      %bech32
+    s++>.f
+    ==
+  ::
+  ++  payment-request
+    |=  payment-request=payment-request:volt
+    ^-  json
+    %-  pairs
+    :~  ['amount-msats' (numb amount-msats.payment-request)]
+        ['payreq' (payreq payreq.payment-request)]
+    ==
+  ::
+  ++  payreq
+    |=  payreq=payreq:volt
+    ^-  json
+    s+payreq
+  ::
+  ++  pay-info
+    |=  info=pay-info:volt
+    ^-  json
+    ~
+  ::
+  ++  hexb
```

Comment:
this may be problem with PSBT encoding - Urbit removes zero padding in hex values, requiring us to use the $wid value here to recreate the correct hex. you can either pass the width value like you are here and use it to construct padding on the JS side or convert to a cord, concat the necessary 0s, and pass as a string coming from the Hoon side

Reply:
++ hexb here is dead code

Reply:
It seems like we don't have %create-funding working in either the dojo (now that psbt it a structure rather than a cord) or in the UI

Reply:
still suspect the problems you've had with the PSBT flow are due to incorrect hex encoding/parsing somewhere, but solve not required for merge